text/000-rustdoc-cfgs-handling.md

```diff
@@ -0,0 +1,283 @@
+    pub struct Bar {
+        pub f: u32,
+    }
+}
+```
+
+As mentioned, both attributes can be used on modules, items and crate root level.
+
+### `#[doc(cfg_hide(...))]`
+
+This attribute is used to prevent some `cfg` to be generated in the visual markers. So in the previous example:
+
+```rust
+#[cfg(any(doc, feature = "futures-io"))]
+pub mod futures {}
+```
+
+It currently displays both `doc` and `feature = "futures-io"` into the documentation, which is not great. To prevent the `doc` cfg to ever be displayed, you can use this attribute at the crate root level:
+
+```rust
+#![doc(cfg_hide(doc))]
+```
+
+Or directly on a given item/module as it covers any of the item's descendants:
+
+```rust
+#[doc(cfg_hide(doc))]
+#[cfg(any(doc, feature = "futures-io"))]
+pub mod futures {
+    // `futures` and all its descendants won't display "doc" in their cfgs.
+}
+```
+
+Then, the `doc` cfg will never be displayed into the documentation.
+
+Rustdoc currently hides `doc` and `doctest` attributes by default and reserves the right to change the list of "hidden by default" attributes.
+
+The attribute accepts only a list of identifiers or key/value items. So you can write:
+
+```rust
+#[doc(cfg_hide(doc, doctest, feature = "something"))]
+#[doc(cfg_hide())]
+```
+
+But you cannot write:
+
+```rust
+#[doc(cfg_hide(not(doc)))]
+```
+
+### `#[doc(cfg_show(...))]`
+
+This attribute does the opposite of `#[doc(cfg_hide(...))]`: if you used `#[doc(cfg_hide(...))]` and want to revert its effect on an item and its descendants, you can use `#[doc(cfg_show(...))]`:
+
+```rust
+#[doc(cfg_hide(doc))]
+#[cfg(any(doc, feature = "futures-io"))]
+pub mod futures {
+    // `futures` and all its descendants won't display "doc" in their cfgs.
+    #[doc(cfg_show(doc))]
+    pub mod child {
+        // `child` and all its descendants will display "doc" in their cfgs.
+    }
+}
+```
+
+The attribute accepts only a list of identifiers or key/value items. So you can write:
+
+```rust
+#[doc(cfg_show(doc, doctest, feature = "something"))]
+#[doc(cfg_show())]
+```
+
+But you cannot write:
+
+```rust
+#[doc(cfg_show(not(doc)))]
+```
+
+## Inheritance
+
+Rustdoc merges `cfg` attributes from parent modules to its children. For example, in this case, the module `non_unix` will describe the entire compatibility matrix for the module, and not just its directly attached information:
+
+```rust
+#[doc(cfg(any(windows, unix)))]
+pub mod desktop {
+    #[doc(cfg(not(unix)))]
+    pub mod non_unix {
+        //
+    }
+}
+```
+
+> ![Available on (Windows or Unix) and non-Unix only.](https://hackmd.io/_uploads/SJrmwYeF2.png)
+
+[Future versions of rustdoc][boolean simplification] may simplify this display down to "available on **Windows** only."
+
+### Re-exports and inlining
+
+`cfg` attributes of a re-export are never merged the re-exported item(s). If `#[doc(inline)]` attribute is used on a re-export, the `cfg` of the re-exported item will be merged with the re-export's.
```

Comment:
issue: sentence two here contradicts the "never" in sentence one.

may also be worth expanding on "merged"

Reply:
Done!